### PR TITLE
Generating ca cert using key vault

### PR DESCRIPTION
**Pull request by phongcao**

### CODE REVIEW

File: deployment/certs/generate-test-certs-keyvault.sh [hunk 1]
```diff
@@ -20,26 +27,37 @@ mkdir $TEMP_DIR
...
+    #########################################################
+
+    # Query an OAuth2 token
+    token=$(curl -X POST https://login.microsoftonline.com/$tenantId/oauth2/token \
+        -d "grant_type=client_credentials&client_id=${clientId}&client_secret=${clientSecret}&resource=https://vault.azure.net" | jq -r '.access_token')
+
+    # Create a new certificate in Key Vault
+    curl -X POST "https://$keyVault.vault.azure.net/certificates/$certName/create?api-version=7.2" \
+        -H "Authorization: Bearer $token" -H "content-type: application/json" \
+        --data "@$certName-cert-policy-curl.json"
```
Comment (Aymalla) [inline deployment/certs/generate-test-certs-keyvault.sh]: we can use the az rest to skip the access token acquiring step
az rest \
     --method post \
     --body @$certName-cert-policy.json \
     --resource "https://vault.azure.net" \
     --headers '{"content-type":"application/json"}' \
     --uri "https://$keyVault.vault.azure.net/certificates/$certName/create" \
     --uri-parameters 'api-version=7.2'

File: deployment/certs/ca-cert-policy-curl.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿{
+  "policy": {
```
Comment (Aymalla) [inline deployment/certs/ca-cert-policy-curl.json]: This file content can be placed inside `ca-cert-policy-curl.json`

File: deployment/certs/generate-test-certs-keyvault.sh (hunk 1 above)
Comment (phongcao) [reply deployment/certs/generate-test-certs-keyvault.sh]: Great feedback, @Aymalla!

File: deployment/certs/ca-cert-policy-curl.json (hunk 2 above)
Comment (phongcao) [reply deployment/certs/ca-cert-policy-curl.json]: Not really, `az rest` requires a slightly different JSON format. If you try creating a new ca file using `ca-cert-policy-curl.json`, you will encounter an error.

**Review by Aymalla — APPROVED** [review]
LGTM

**Pull request merged**